openedx/openedx-platform
# feat: add PREPEND_LOCALE_PATHS configuration setting

## Description


edx-platform currently supports COMPREHENSIVE_THEME_LOCALE_PATHS setting, which appends paths to the end of LOCALE_PATHS, but there's currently no way to add additional paths to the start of the list to override the default translations.

In this PR, we added the PREPEND_LOCALE_PATHS setting to achieve this.

- Affects the Operator

## Supporting information

- PR for the `configuration` repo. https://github.com/openedx/configuration/pull/6677
- [SE-5299](https://tasks.opencraft.com/browse/SE-5299)
- https://pr29851.sandbox.opencraft.hosting/

## Testing instructions

On the docker devstack:
- `make dev.shell.lms`
- Edit the file `/edx/etc/lms.yml`. Add a `PREPEND_LOCALE_PATHS` setting like
```yaml
PREPEND_LOCALE_PATHS: [/edx/src/]
```
- Copy a translation file into that folder.
```bash
cp -r /edx/app/edxapp/edx-platform/conf/locale/fr/ /edx/src/es
```
Note that this is copying French to Spanish folder. It makes seeing the differences easier.
- In your browser load the LMS, and change the `openedx-language-cookie`'s value to `es`.
- Restart your lms dev server.
- If done correctly you should get the french translations loaded (it sometimes takes 5 minutes to load).

## Deadline

"None"

## Other information

**Settings**
```yaml
configuration_version: keith/se_5299_prepend_local_paths
configuration_source_repo_url: https://github.com/open-craft/configuration
edx_platform_repository_url: https://github.com/open-craft/edx-platform.git
```
